### DISCUSSION TITLE
can I call `useRouter()` inside store?

### DISCUSSION THREAD

**Discussion by husayt** [vuejs/pinia#442] [Help and Questions] [answered]
Hi, can I call `useRouter()` inside store or shall i pass router as a param?

I tried calling `useRouter()` from inside action and it returned undefined. I am not sure if I am missing something or is wrong practice to start with?

Thanks in advance

**Comment by posva** [comment, marked as answer]
Like any function based on `inject()`, you won't be able to use it outside of `setup()`. With the plugin API you can add a router property to each store:

```js
pinia.use(() => ({ router })

createApp(App).use(router).use(pinia).mount('#app')
```

And type it with

```ts
// extensions.d.ts
import 'pinia'
import { Router } from 'vue-router'

declare module 'pinia' {
  export interface PiniaCustomProperties {
    router: Router
  }
}
```

The plugin API is a WIP at #416 
The ongoing docs are at https://deploy-preview-416--pinia.netlify.app/core-concepts/plugins.html#typescript
